api/src/main/java/com/epam/esm/controller/GiftController.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.model.CustomSearchRequest;
+import com.epam.esm.model.dto.GiftCertificateDto;
+import com.epam.esm.service.GiftService;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/gifts")
+public class GiftController {
+    private final GiftService giftService;
+    private final CustomSearchRequest defaultCustomSearchRequest;
+
+
+    @Autowired
+    public GiftController(GiftService giftService) {
+        this.giftService = giftService;
+        defaultCustomSearchRequest = CustomSearchRequest.builder().build();
+    }
+
+
+    @GetMapping
+    @ApiOperation(value = "Api v1. Get all gifts")
+    public ResponseEntity<List<GiftCertificateDto>> allGifts(
+            @ApiParam(name = "descriptionPrefix", value = "Search by gift description") String descriptionPrefix,
```

Comment:
Don't you need to annotate all these parameters with @RequestParam?

Reply:
I missed it here, but according spring documentation all arguments which don't need argument resolver, is treated as if it were annotated with @RequestParam.